packages/snjs/mocha/features.test.js

```diff
@@ -226,4 +230,29 @@ describe('features', () => {
       expect(await application.getSensitiveSetting(SettingName.ExtensionKey)).to.equal(true);
     });
   });
+
+  describe('full sync completed observer', () => {
+    it('should update extension key user setting when local data is loaded and user is signed in', async () => {
+      expect(await application.getSensitiveSetting(SettingName.ExtensionKey)).to.equal(false);
+      const extensionKey = Uuid.GenerateUuidSynchronously().split('-').join('');
+      application = await Factory.signOutApplicationAndReturnNew(application);
+      sinon.stub(application.itemManager, 'getItems').callsFake(() => {
+        return [
+          {
```

Comment:
The object being returned here is not really an item. It's missing content_type. So how does the ExtensionRepo observer in features_service pick this up?

Reply:
This replaces the actual getItems method with a method that always returns this, the idea is just to mock a situation where there's a pre-existing extension repo item

Reply:
Right but the observers are notified based on the content_type of the item

Reply:
I've now refactored it so that it takes contentType into account to make it more similar to the actual method

Reply:
So it was working without it? Weird.

Reply:
> Right but the observers are notified based on the content_type of the item

That's for the itemManager observer, which is the test right above this one and doesn't use a mock. This one is for the FullSyncCompleted observer